src/diffpy/utils/scattering_objects/diffraction_object.py

```diff
@@ -0,0 +1,492 @@
+import datetime
+from copy import deepcopy
+
+import numpy as np
+
+from diffpy.utils.tools import get_package_info
+
+QQUANTITIES = ["q"]
+ANGLEQUANTITIES = ["angle", "tth", "twotheta", "2theta"]
+DQUANTITIES = ["d", "dspace"]
+XQUANTITIES = ANGLEQUANTITIES + DQUANTITIES + QQUANTITIES
+XUNITS = ["degrees", "radians", "rad", "deg", "inv_angs", "inv_nm", "nm-1", "A-1"]
+
+x_grid_emsg = (
+    "objects are not on the same x-grid. You may add them using the self.add method "
+    "and specifying how to handle the mismatch."
+)
+
+
+class DiffractionObject:
```

Comment:
A new file for `DiffractionObject`. File name changed from `diffraction_objects` to `diffraction_object`